### ISSUE TITLE
Add Mailosaur to tests

### ISSUE THREAD

**Issue by peterjaap** [elgentos/magento2-cypress-testing-suite#16]
For some tests like reset password nieuws, newsletter subscription, order confirmation etc mails need te be received and read. For this purpose [Mailosaur](https://mailosaur.com/) can be used. A mailosaur API-key and account will be needed, the API-key needs to be added to the `cypress.json` "env" object or to a `cypress.env.json` file. And the serverId needs to be added to the tests where mailosaur is used.

```json
// cypress.json
"env": {
    "MAILOSAUR_API_KEY": "some-hash"
}
```

```js
// sometest.spec.js
const serverId = 'SERVER_ID';
const testMail = `something@${serverId}.mailosaur.net`;

cy.mailodaurGetMessage(serverId, {
    sentTo: testEmail
}).then(email => {
    // Do stuff
})
```

- [ ] Install mailosaur
- [ ] Make sure cypress can read sent mails
- [ ] Make sure it is optional since it requires an extra dependency

## Tests
- [ ] A confirmation email will be sent after placing the order 
- [ ] If payment is made an invoice will be sent 

Example:

```js
it('can receive a confirmation mail that an order has been placed', () => {
    // TODO: implement mailosaur
})

it('can receive an invoice through the mail when a payment is completed', () => {
    // TODO: implement mailosaur
})
```

**Comment by AndrewRMillar** [comment]
Will not work when testing locally and also not on our testing site, because of the k8s setup. Needs to use a different solution than our current testing setup.

**Comment by kolaente** [comment]
Another interesting solution might be [mailhog](https://github.com/mailhog/MailHog) - it has an api and you can host it locally in a docker container.

**Comment by peterjaap** [comment]
@kolaente yes we use that locally and that works fine, but when running the tests in a CI/CD situation it's not useful. Using Mailosaur will enable us to inspect the emails later when something went wrong.

**Comment by peterjaap** [comment]
@norgeindian I haven't run in that yet, maybe because we generally use `mageplaza/module-smtp` to send mails with and we configure that to use some SMTP. I guess that extension doesn't rely on sendmail under the hood.

**Comment by norgeindian** [comment]
@peterjaap , thanks for this amazingly fast reply.
Good idea, that is surely the best way. I included mailhog now and set up a similar extension to use the mailhog SMTP port. That seems to work.
